enhancements/network/service-egress-traffic-steering.md

```diff
@@ -0,0 +1,284 @@
+
+With all of the above implemented, a user can follow these steps to create a functioning LoadBalancer service whose endpoints exit the cluster with its IP.
+
+1. Create the IPAdressPool with the desired IP for the service. It makes sense to set `autoAssign: false` so it is not taken by another service - our service will request that pool explicitly. 
+```yaml
+apiVersion: metallb.io/v1beta1
+kind: IPAddressPool
+metadata:
+  name: example-pool
+  namespace: metallb-system
+spec:
+  addresses:
+  - 172.19.0.100/32
+  autoAssign: false
+```
+
+2. Create the LoadBalancer service. We create it with 2 annotations:
+- `metallb.universe.tf/address-pool` - to explicitly request the IP to be from the `example-pool`.
+- `k8s.ovn.org/egress-service` - to request that all of the endpoints of the service exit the cluster with the service's IP. We also provide a `network-hint` so that the traffic exits from an interface that belongs to it.
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: example-service
+  namespace: some-namespace
+  annotations:
+    metallb.universe.tf/address-pool: example-pool
+    k8s.ovn.org/egress-service: '{"network-hint":172.19.0.0/24}'
+spec:
+  selector:
+    app: example
+  ports:
+    - name: http
+      protocol: TCP
+      port: 8080
+      targetPort: 8080
+  type: LoadBalancer
+```
+
+3. Advertise the service from the node in charge of the service's traffic. So far the service is "broken" - it is not reachable from outside the cluster and if the pods try to send traffic outside it would probably not come back as it is SNATed to an IP which is unknown.
+We create the advertisements targetting only the node that is in charge of the service's traffic using the `nodeSelectors` field, relying on ovn-k to label the node properly.
+```yaml
+apiVersion: metallb.io/v1beta1
+kind: L2Advertisement
+metadata:
+  name: example-l2-adv
+  namespace: metallb-system
+spec:
+  ipAddressPools:
+  - example-pool
+  nodeSelectors:
+  - matchLabels:
+      k8s.ovn.org/example-service-some-namespace-host: ""
+---
+apiVersion: metallb.io/v1beta1
+kind: BGPAdvertisement
+metadata:
+  name: example-l2-adv
+  namespace: metallb-system
+spec:
+  ipAddressPools:
+  - example-pool
+  nodeSelectors:
+  - matchLabels:
+      k8s.ovn.org/example-service-some-namespace-host: ""
+```
+While possible to create more advertisements resources for the `example-pool`, it is the user's responsibility to make sure that the pool is advertised only by advertisements targetting the node holding the `k8s.ovn.org/<svc-name>-<svc namespace>-host: ""` label - otherwise the traffic of the service will be broken.
+### User Stories
+
+#### Story 1
+
+As a user of OpenShift, I should be able to have functioning LoadBalancer services whose backing pods exit the cluster with the service's IP from an interface that belongs to a network I specify.
+
+### API Extensions
+A new annotation `k8s.ovn.org/egress-service` is supported for LoadBalancer services.
```

Comment:
I think this section is only for stuff that touches `openshift/api`, so new annotations wouldn't count

Reply:
removed